src/typecode/magic2.py

```diff
@@ -74,24 +79,184 @@
+        if failover_lib:
+            warnings.warn(
+                'System libmagic found in typical location is used. '
+                'Install instead a typecode-libmagic plugin for best support.'
+            )
+            return failover_lib
+
+    # try the PATH
+    if not dll_loc:
+        dll = 'libmagic.dll' if on_windows else 'libmagic.so'
+        dll_loc = command.find_in_path(dll)
+
+        if dll_loc:
+            warnings.warn(
+                'libmagic found in the PATH. '
+                'Install instead a typecode-libmagic plugin for best support.'
+            )
+
+        if TRACE and dll_loc:
+            logger_debug('load_lib:', 'got path magic location:', dll_loc)
+
+    if not dll_loc or not os.path.isfile(dll_loc):
+        raise NoMagicLibError(
+            'CRITICAL: libmagic DLL and its magic database are not installed. '
             'Unable to continue: you need to install a valid typecode-libmagic '
-            'plugin with a valid and proper libmagic and magic DB available.'
+            'plugin with a valid and proper libmagic and magic DB available.\n'
+            f'OR set the {TYPECODE_LIBMAGIC_PATH_ENVVAR} and '
+            f'{TYPECODE_LIBMAGIC_DB_PATH_ENVVAR} environment variables.\n'
+            f'OR install libmagic in typical common locations.\n'
+            f'OR have a libmagic in the system PATH.\n'
     )
-    return command.load_shared_library(dll, libdir)
+    return command.load_shared_library(dll_loc)
+
+
+def get_magicdb_location(_cache=[]):
+    """
+    Return the location of the magicdb loaded from either:
+    - an environment variable ``TYPECODE_LIBMAGIC_DB_PATH``,
+    - a plugin-provided path,
+    - the system PATH.
+    Trigger a warning if no magicdb file is found.
+    """
+    if _cache:
+        return _cache[0]
+
+    from plugincode.location_provider import get_location
+
+    # try the environment first
+    magicdb_loc = os.environ.get(TYPECODE_LIBMAGIC_DB_PATH_ENVVAR)
+
+    if TRACE and magicdb_loc:
+        logger_debug('get_magicdb_location:', 'got environ magicdb location:', magicdb_loc)
+
+    # try a plugin-provided path second
+    if not magicdb_loc:
+        magicdb_loc = get_location(TYPECODE_LIBMAGIC_DB)
+
+        if TRACE and magicdb_loc:
+            logger_debug('get_magicdb_location:', 'got plugin magicdb location:', magicdb_loc)
+
+    # try the PATH
+    if not magicdb_loc:
+        db = 'magic.mgc'
+        magicdb_loc = command.find_in_path(db)
+
+        if magicdb_loc:
+            warnings.warn(
+                'magicdb found in the PATH. '
+                'Install instead a typecode-libmagic plugin for best support.\n'
+                f'OR set the {TYPECODE_LIBMAGIC_DB_PATH_ENVVAR} environment variable.'
+            )
+
+        if TRACE and magicdb_loc:
+            logger_debug('get_magicdb_location:', 'got path magicdb location:', magicdb_loc)
+
+    if not magicdb_loc:
+        warnings.warn(
```

Comment:
I think this not needed as libmagic will (probably) already complain if it does not find its db in standard location.

Reply:
@tardyp you wrote:

> Franckly I am not a huge fan of vendoring for opensource projects, although I do understand the need to limit dependency variablility for support reason

I agree and the Pygments vendoring here is a must here as we have dependencies on specific versions and vendoring is the sane approach to avoid conflicts when this fairly common library is installed otherwise.

Reply:
Good point... I like keeping the warning on our side too though as the message would be more explicit